dashboard/test/controllers/script_levels_controller_test.rb

```diff
@@ -58,6 +58,37 @@ class ScriptLevelsControllerTest < ActionController::TestCase
     @pilot_teacher = create :teacher, pilot_experiment: 'pilot-experiment'
     pilot_section = create :section, user: @pilot_teacher, script: pilot_script
     @pilot_student = create(:follower, section: pilot_section).student_user
+
+    # Some of the functionality we're testing here is data-specific,
+    # and needs scripts with certain names to work. In the old
+    # fixture-based model, this data was all provided; in the new
+    # factory-based model, we need to do a little prep.
+    tested_script_names = [
+      'ECSPD',
+      Script::FLAPPY_NAME,
+      Script::FROZEN_NAME,
+      Script::HOC_NAME,
+      Script::PLAYLAB_NAME
+    ]
+
+    tested_script_names.each do |script_name|
+      script = Script.find_by_name(script_name)
+
+      # create the Script if we don't have one already
+      unless script.present?
+        script = create(:script, :with_levels, levels_count: 5, name: script_name)
```

Comment:
note that this is creating a generic script rather than one that exactly matches the internal levels/script_levels/lessons structure of the script it's standing in for. That should generally be fine; we have a very small number of tests that actually care about the content within these scripts, but I haven't yet come across any that care about the structure of the content and even if I do I'm confident we'll be able to update them to work more generically.